Pi_Code/Controller_Operations.py

```diff
@@ -0,0 +1,113 @@
+from multiprocessing import Process
+import pygame
+
+# Axes legend:
+# Axes 0: Roll (right is positive, left is negative)
+# Axes 1: Pitch (back is positive, forward is negative)
+# Axes 2: Throttle (0% position is positive, 100% position is negative)
+# Axes 3: Yaw (clockwise is positive, counterclockwise is negative)
+# Button legend:
+# Button 0: Trigger
+# Button 6: Labelled as 07 on the joystick
+# Button 5: Labelled as 06 on the joystick
+# Button 1: Labelled as 02 on the joystick
+
+
+class ControllerOps:
+
+    def __init__(self, dataFile, connection, SaveSensorData):
+        # We need to display a window in order to read joystick inputs.
+        pygame.joystick.init()
+        screen = pygame.display.set_mode((10, 10))
+
+        self.controller = pygame.joystick.Joystick(0)
+        self.controller.init()
+
+        self.idle = False # TODO: Change idle to true and add state switching in the main Submarine script
+        # Currently we'll just skip right past the idle stage and right into the auto sub stage. Idle is intended to be
+		# where we click the "start" button on the joystick to begin the operation of the sub
+        self.read_stick = True
+        self.autosub = True
+
+        # Get the properties of the joystick
+        # Not necessary currently, may be useful later
+        print("Found joystick: " + self.controller.get_name())
+        buttons = self.controller.get_numbuttons()
```

Comment:
Why are you making these calls if you are not using their values?

Reply:
Removed.